src/adapter/src/catalog/apply.rs

```diff
@@ -344,6 +504,113 @@ impl CatalogState {
         );
     }
 
+    /// Applies an item update to `self`.
+    ///
+    /// Returns a `GlobalId` on success, if the applied update added a new `GlobalID` to `self`.
+    /// Returns a dependency on failure, if the update could not be applied due to a missing
+    /// dependency.
+    #[instrument(level = "debug")]
+    fn apply_item_update(
+        &mut self,
+        item: mz_catalog::durable::Item,
+        diff: Diff,
+    ) -> Result<Option<GlobalId>, ApplyUpdateError> {
```

Comment:
@sploiselle This PR will probably cause a conflict with your PR https://github.com/MaterializeInc/materialize/pull/26556/files. You just have to move your comment from `open.rs` to right here and the conflicts should resolve.